src/main/java/server/Process.java

```diff
@@ -0,0 +1,5 @@
+package server;
+
+public interface Process {
+    void runWith(ConnectionDataStream client);
```

Comment:
Why is the variable called `client`?

Reply:
Would a better name be user-agent or just agent?

Reply:
I see. Then should the interface be called `Client`? It is a little fishy to me that you are depending on an abstraction here with `ConnectionDataStream`, but the argument name nods to the concrete implementation (`ConnectedClient`). What this indicates to me is that the language you are using to represent the domain is quite flushed out completely.

Reply:
Ok, so i think im starting to follow. Is the issue the variable name 'client'? That it implies the only connection will be a client connection rather than a generic connected application/device/w.e?

If so, i may have misunderstood earlier.